3.5/troubleshooting-reduce-memory-footprint.md

```diff
@@ -0,0 +1,353 @@
+the read buffers if you use graph features.
+
+Please keep in mind that during query execution additional memory will be used
+for query results temporarily. If you are tight on memory, you may want to go
+for 7 GB each instead.
+
+If you have an embedded system or your development laptop, you can use all of
+the above settings to lower the memory footprint further. For normal operation,
+especially production, these settings are not recommended.
+
+Linux System Configuration
+--------------------------
+
+The main deployment target for ArangoDB is Linux. As you’ve learned above
+ArangoDB and its innards work a lot with memory. Thus its vital to know how
+ArangoDB and the Linux kernel interact on that matter. The linux kernel offers
+several modes of how it will manage memory. You can influence this via the proc
+filesystem, the file `/etc/sysctl.conf` or a file in `/etc/sysctl.conf.d/` which
+your system will apply to the kernel settings at boot time. The settings as
+named below are intended for the sysctl infrastructure, meaning that they map
+to the `proc` filesystem as `/proc/sys/vm/overcommit_memory`.
+
+After studying the kernel documentation our former recommendation was to set
+`vm.overcommit_memory` to **2** – which we (and others) found out is **not good
+in all situations**. It seems that there is an issue with this setting in
+combination with the bundled memory allocator ArangoDB ships with (jemalloc) in
+some environments.
+
+The allocator demands consecutive blocks of memory from the kernel, which are
+also mapped to on-disk blocks. This is done on behalf of the server process
+(*arangod*). The process may use some chunks of a block for a long time span, but
+others only for a short while and therefore release the memory. It is then up to
+the allocator to return the freed parts back to the kernel. Because it can only
+give back consecutive blocks of memory, it has to split the large block into
+multiple small blocks and can then return the unused ones.
+
+With an `vm.overcommit_memory` kernel settings value of **2**, the allocator may
+have trouble with splitting existing memory mappings, which makes the *number*
+of memory mappings of an arangod server process grow over time. This can lead to
+the kernel refusing to hand out more memory to the arangod process, even if more
+physical memory is available. The kernel will only grant up to `vm.max_map_count`
+memory mappings to each process, which defaults to 65530 on many Linux
+environments.
+
+Another issue when running jemalloc with `vm.overcommit_memory` set to **2** is
+that for some workloads the amount of memory that the Linux kernel tracks as
+“*committed memory*” also grows over time and never decreases. Eventually,
+*arangod* may not get any more memory simply because it reaches the configured
+overcommit limit (physical RAM * `overcommit_ratio` + swap space).
+
+The solution is to modify the value of `vm.overcommit_memory` from **2** to
```

Comment:
no link to the section of the guide where we also talk about the topic https://www.arangodb.com/docs/3.4/installation-linux-osconfiguration.html#over-commit-memory